geopandas/tests/test_pandas_methods.py

```diff
@@ -0,0 +1,175 @@
+from shapely.geometry import Point
+
+from geopandas import GeoDataFrame, GeoSeries
+from geopandas.tests.util import assert_geoseries_equal
+
+
+@pytest.fixture
+def s():
+    return GeoSeries([Point(x, y) for x, y in zip(range(3), range(3))])
+
+
+@pytest.fixture
+def df():
+    return GeoDataFrame({'geometry': [Point(x, x) for x in range(3)],
+                         'value1': range(3), 'value2': [1, 2, 1]})
+
+
+def test_repr(s, df):
+    assert 'POINT' in repr(s)
+    assert 'POINT' in repr(df)
+
+
+def test_indexing(s, df):
+
+    # accessing scalar from the geometry (colunm)
+    exp = Point(1, 1)
+    assert s[1] == exp
+    assert s.loc[1] == exp
+    assert s.iloc[1] == exp
+    assert df.loc[1, 'geometry'] == exp
+    assert df.iloc[1, 0] == exp
+
+    # multiple values
+    exp = GeoSeries([Point(2, 2), Point(0, 0)], index=[2, 0])
+    assert_geoseries_equal(s.loc[[2, 0]], exp)
+    assert_geoseries_equal(s.iloc[[2, 0]], exp)
+    assert_geoseries_equal(s.reindex([2, 0]), exp)
+    assert_geoseries_equal(df.loc[[2, 0], 'geometry'], exp)
+    # TODO here iloc does not return a GeoSeries
+    assert_series_equal(df.iloc[[2, 0], 0], exp, check_series_type=False,
+                        check_names=False)
+
+    # boolean indexing
+    exp = GeoSeries([Point(0, 0), Point(2, 2)], index=[0, 2])
+    mask = np.array([True, False, True])
+    assert_geoseries_equal(s[mask], exp)
+    assert_geoseries_equal(s.loc[mask], exp)
+    assert_geoseries_equal(df[mask]['geometry'], exp)
+    assert_geoseries_equal(df.loc[mask, 'geometry'], exp)
+
+
+def test_assignment(s, df):
+    exp = GeoSeries([Point(10, 10), Point(1, 1), Point(2, 2)])
+
+    s2 = s.copy()
+    s2[0] = Point(10, 10)
+    assert_geoseries_equal(s2, exp)
+
+    s2 = s.copy()
+    s2.loc[0] = Point(10, 10)
+    assert_geoseries_equal(s2, exp)
+
+    s2 = s.copy()
+    s2.iloc[0] = Point(10, 10)
+    assert_geoseries_equal(s2, exp)
+
+    df2 = df.copy()
+    df2.loc[0, 'geometry'] = Point(10, 10)
+    assert_geoseries_equal(df2['geometry'], exp)
+
+    df2 = df.copy()
+    df2.iloc[0, 0] = Point(10, 10)
+    assert_geoseries_equal(df2['geometry'], exp)
+
+
+def test_astype(s):
+
+    with pytest.raises(TypeError):
+        s.astype(int)
+
+    assert s.astype(str)[0] == 'POINT (0 0)'
+
+
+def test_to_csv(df):
+
+    exp = ('geometry,value1,value2\nPOINT (0 0),0,1\nPOINT (1 1),1,2\n'
+           'POINT (2 2),2,1\n')
+    assert df.to_csv(index=False) == exp
+
+
+def test_numerical_operations(s, df):
+
+    # df methods ignore the geometry column
+    exp = pd.Series([3, 4], index=['value1', 'value2'])
+    assert_series_equal(df.sum(), exp)
+
+    # series methods raise error
+    with pytest.raises(TypeError):
+        s.sum()
+
+    with pytest.raises(TypeError):
+        s.max()
+
+    with pytest.raises(TypeError):
+        s.idxmax()
+
+    # numerical ops raise an error
+    with pytest.raises(TypeError):
+        df + 1
```

Comment:
Might want to verify that the `.numeric_types` attribute (or whatever it is called that collects only numeric dtypes) works